cloudflare/resource_cloudflare_custom_ssl.go

```diff
@@ -0,0 +1,337 @@
+						},
+						"geo_restrictions": {
+							Type:     schema.TypeMap,
+							Optional: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"label": {
+										Type:         schema.TypeString,
+										Optional:     true,
+										ValidateFunc: validation.StringInSlice([]string{"us", "eu", "highest_security"}, false),
+									},
+								},
+							},
+						},
+						"type": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							ValidateFunc: validation.StringInSlice([]string{"legacy_custom", "sni_custom"}, false),
+						},
+					},
+				},
+			},
+			"hosts": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"issuer": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"signature": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"uploaded_on": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"modified_on": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"expires_on": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"priority": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceCloudflareCustomSslCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudflare.API)
+	zoneID := d.Get("zone_id").(string)
+	log.Printf("[DEBUG] zone ID: %s", zoneID)
+	zcso, err := expandToZoneCustomSSLOptions(d)
+	if err != nil {
+		return fmt.Errorf("Failed to create custom ssl cert: %s", err)
+	}
+
+	res, err := client.CreateSSL(zoneID, zcso)
+	if err != nil {
+		return fmt.Errorf("Failed to create custom ssl cert: %s", err)
+	}
+
+	if res.ID == "" {
+		return fmt.Errorf("Failed to find custom ssl in Create response: id was empty")
+	}
+
+	d.SetId(res.ID)
+
+	log.Printf("[INFO] Cloudflare Custom SSL ID: %s", d.Id())
+
+	return resourceCloudflareCustomSslRead(d, meta)
+}
+
+func resourceCloudflareCustomSslUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudflare.API)
+	zoneID := d.Get("zone_id").(string)
+	certID := d.Id()
+	var updateErr bool = false
+	var reprioritizeErr bool = false
+	log.Printf("[DEBUG] zone ID: %s", zoneID)
+
+	zcso, err := expandToZoneCustomSSLOptions(d)
+	if err != nil {
+		return fmt.Errorf("Failed to update custom ssl cert: %s", err)
+	}
+
+	res, uErr := client.UpdateSSL(zoneID, certID, zcso)
+	if uErr != nil {
+		fmt.Errorf("Failed to update custom ssl cert: %s", uErr)
+		updateErr = true
+	} else {
+		log.Printf("[DEBUG] Custom SSL set to: %s", res.ID)
+	}
+
+	zcsp, err := expandToZoneCustomSSLPriority(d)
+	if err != nil {
+		fmt.Printf("Failed to update custom ssl cert: %s", err)
+	}
+
+	resList, reErr := client.ReprioritizeSSL(zoneID, zcsp)
```

Comment:
Should use partial state mode here: https://www.terraform.io/docs/plugins/provider.html#resource-data